redisai/model.go

```diff
@@ -17,58 +18,60 @@ type ModelInterface interface {
 	SetDevice(device string)
 	Backend() string
 	SetBackend(backend string)
+	Tag() string
+	SetTag(tag string)
 }
 
 func modelSetFlatArgs(keyName, backend, device string, inputs, outputs []string, blob []byte) redis.Args {
-	args := redis.Args{}.Add(keyName, backend, device)
-	if len(inputs) > 0 {
-		args = args.Add("INPUTS").AddFlat(inputs)
-	}
-	if len(outputs) > 0 {
-		args = args.Add("OUTPUTS").AddFlat(outputs)
-	}
-	args = args.Add("BLOB")
-	args = args.Add(blob)
-	return args
+	modelInterface := implementations.NewModel(backend, device)
+	modelInterface.SetInputs(inputs)
+	modelInterface.SetOutputs(outputs)
+	modelInterface.SetBlob(blob)
+	return modelSetInterfaceArgs(keyName, modelInterface)
 }
 
 func modelSetInterfaceArgs(keyName string, modelInterface ModelInterface) redis.Args {
-	return modelSetFlatArgs(keyName, modelInterface.Backend(), modelInterface.Device(), modelInterface.Inputs(), modelInterface.Outputs(), modelInterface.Blob())
-}
-
-func modelRunFlatArgs(name string, inputTensorNames, outputTensorNames []string) redis.Args {
-	args := redis.Args{}
-	args = args.Add(name)
-	if len(inputTensorNames) > 0 {
-		args = args.Add("INPUTS").AddFlat(inputTensorNames)
+	args := redis.Args{keyName}
+	if len(modelInterface.Backend()) > 0 {
+		args = args.Add(modelInterface.Backend())
+	}
+	if len(modelInterface.Device()) > 0 {
+		args = args.Add(modelInterface.Device())
+	}
+	if len(modelInterface.Tag()) > 0 {
+		args = args.Add("TAG", modelInterface.Tag())
 	}
-	if len(outputTensorNames) > 0 {
-		args = args.Add("OUTPUTS").AddFlat(outputTensorNames)
+	if len(modelInterface.Inputs()) > 0 {
+		args = args.Add("INPUTS").AddFlat(modelInterface.Inputs())
+	}
+	if len(modelInterface.Outputs()) > 0 {
+		args = args.Add("OUTPUTS").AddFlat(modelInterface.Outputs())
+	}
+	if modelInterface.Blob() != nil {
+		args = args.Add("BLOB", modelInterface.Blob())
 	}
 	return args
 }
 
-func modelGetParseToInterface(reply interface{}, model ModelInterface) (err error) {
-	var backend string
-	var device string
-	var blob []byte
-	err, backend, device, blob = modelGetParseReply(reply)
-	if err != nil {
-		return err
-	}
-	model.SetBackend(backend)
-	model.SetDevice(device)
-	model.SetBlob(blob)
-	return
+func modelRunFlatArgs(name string, inputTensorNames, outputTensorNames []string) redis.Args {
```

Comment:
this should not work given that we're mixing modelset and modelrun args. please see the correction on https://github.com/dengliming/redisai-go/pull/1